compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -2458,8 +2460,11 @@ fn param_env_with_gat_bounds<'tcx>(
 
     for impl_ty in impl_tys_to_install {
         let trait_ty = match impl_ty.container {
+            ty::AssocItemContainer::InherentImpl => bug!(),
             ty::AssocItemContainer::Trait => impl_ty,
-            ty::AssocItemContainer::Impl => tcx.associated_item(impl_ty.trait_item_def_id.unwrap()),
+            ty::AssocItemContainer::TraitImpl => {
+                tcx.associated_item(impl_ty.trait_item_def_id.unwrap())
```

Comment:
This is a sleeping ICE, because `trait_item_def_id` can indeed be `None` if there are compilation errors.

Reply:
Now this is `Err(_) => continue,`